users/views/_admin_permissions.php

```diff
@@ -145,16 +145,19 @@
   <div class="col-12 col-md-6">
     <h5>Manage the Permission Level</h5>
     <?php
+    if (empty($manage)) {
+      $manage = 0;
+    }
     $q = $db->query("SELECT * FROM permissions WHERE id = ?", [$manage]);
     $c = $q->count();
     if ($c > 0) {
       $perm = $q->first();
-      $usersQ = $db->query("SELECT u.id,u.fname, u.lname, u.email,
-            CASE WHEN p.permission_id is null THEN 0 ELSE 1 END AS hasPerm
-            FROM users AS u
-            LEFT JOIN user_permission_matches AS p ON p.user_id = u.id AND p.permission_id = ?
-            GROUP BY u.id
-            ", [$manage]);
+      $usersQ = $db->query("SELECT u.id, u.fname, u.lname, u.email,
+      CASE WHEN p.permission_id is null THEN 0 ELSE 1 END AS hasPerm
+      FROM users AS u
+      LEFT JOIN user_permission_matches AS p ON p.user_id = u.id AND p.permission_id = ?
+      GROUP BY u.id, hasPerm
+        ", [$manage]);
```

Comment:
It was giving an error without putting other columns. I searched the error on google and if I remember correctly it had something to do with postgresql using it differently than mysql so thats why I had to add this column to fix the error otherwise it was not listing the permissions.

Reply:
Can you please hunt down and link the mentioned resource on the problem and the solution? I'm not saying this is wrong, just curious about the exact nature of the problem.

Reply:
This is the exact error that I was getting:

`ERROR: SQLSTATE[42803]: Grouping error: 7 ERROR: column "p.permission_id" must appear in the GROUP BY clause or be used in an aggregate function LINE 2: CASE WHEN p.permission_id is null THEN 0 ELSE 1 END AS... ^`

So I googled it, and it said that you are trying to `GROUP BY` a column that is not included in the `SELECT` clause, which is `hasPerm` in our case.

I got the idea from this post:

[StackoverFlow](https://stackoverflow.com/questions/69821563/error-column-must-appear-in-the-group-by-clause-or-be-used-in-an-aggregate-func)

Since the `hasPerm` column is used in the `SELECT` query and not in the `GROUP BY` therefore, I included it, and the problem was solved.

Reply:
Makes sense, thank you! Please be aware of this in future works with GROUP BY clauses.